/**
 * Copyright (c) 2022 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";
package proto_panda;

import "assemblyFunction.proto";
import "assemblyRecord.proto";
import "assemblyType.proto";
import "assemblyLiterals.proto";

message Program {
    message RecordUnit  {
        bytes key = 1;
        Record value = 2;
    }
    message FunctionUnit  {
        bytes key = 1;
        Function value = 2;
    }
    message FunctionSynnoym  {
        bytes key = 1;
        repeated bytes value = 2;
    }
    message LiteralUnit {
        bytes key = 1;
        LiteralArray value = 2;
    }
    uint32 lang = 1;
    repeated RecordUnit record_table = 2;
    repeated FunctionUnit function_table = 3;
    repeated FunctionSynnoym function_synonyms = 4;
    repeated LiteralUnit literalarray_table = 5;
    repeated bytes strings = 6;
    repeated Type array_types = 7;
}
